common/src/main/java/com/microsoft/identity/common/internal/request/MsalBrokerRequestAdapter.java

```diff
@@ -527,6 +528,40 @@ public Bundle getRequestBundleForGenerateShr(@NonNull final GenerateShrCommandPa
         return requestBundle;
     }
 
+
+    /**
+     * Method to construct a request bundle for getting device id for given tenant if device is registered
+     * in that tenant.
+     * @param parameters                     input parameters of type {@link GetAadDeviceIdCommandParameters}
+     * @param negotiatedBrokerProtocolVersion protocol version established by broker hello.
+     * @return request Bundle
+     */
+    public Bundle getRequestBundleForAadDeviceIdRequest(
+            @NonNull final GetAadDeviceIdCommandParameters parameters,
+            @Nullable final String negotiatedBrokerProtocolVersion
+    ) {
+        final String methodTag = TAG + ":getRequestBundleGetAadDeviceIdRequest";
+        Logger.info(methodTag, "Constructing result bundle from getRequestBundleGetAadDeviceIdRequest.");
+
+        final BrokerRequest brokerRequest = BrokerRequest.builder()
```

Comment:
Do we need all this? can we just send correlation id + tenant id? would that be enough?
(no need to use/serialize/deserialize broker request object)

Reply:
I want redirect to enforce check on the caller. And correlationId for tracking. Other parameters are like additional/metadata which can be helpful when I add telemetry etc